addons/sales_team/models/crm_team.py

```diff
@@ -80,6 +83,9 @@ def _get_default_team_id(self, user_id=None, domain=None):
 
         if not team:
             teams = self.env['crm.team'].search([('company_id', 'in', valid_cids)])
+            if self.env.company in teams.company_id:
+                # Prefer the current company if possible
+                teams = teams.filtered(lambda t: t.company_id == self.env.company)
```

Comment:
We probably want to apply the domain first, otherwise we could eliminate the only teams that would match it.